aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationOutbox.java

```diff
@@ -38,165 +39,152 @@
-            // Convert the storageItemChange (that we want to store) into a record
-            StorageItemChange.Record record = storageItemChange.toRecord(storageItemChangeConverter);
+        // defer() the creation of a Completable, until someone subscribes enqueue().
+        // When they do, create() a Completable that wraps a save() call to LocalStorageAdapter.
+        return Completable.defer(() -> Completable.create(subscriber -> {
+            // Convert the PendingMutation (that we want to store) into a Record
+            PendingMutation.PersistentRecord record = converter.toRecord(pendingMutation);
             // Save it.
-            localStorageAdapter.save(record, StorageItemChange.Initiator.SYNC_ENGINE,
-                recordOfRecord -> {
+            storage.save(record, StorageItemChange.Initiator.SYNC_ENGINE,
+                saved -> {
                     // The return value is a record that we saved a record.
                     // So, we would have to "unwrap" it, to get the item we saved, out.
                     // Forget that. We know the save succeeded, so just emit the
                     // original thing enqueue() got as a param.
-                    pendingStorageItemChanges.onNext(storageItemChange);
-                    subscriber.onSuccess(storageItemChange);
+                    LOG.info("Successfully enqueued " + pendingMutation);
+                    pendingMutations.onNext(pendingMutation);
+                    subscriber.onComplete();
                 },
                 error -> {
-                    pendingStorageItemChanges.onError(error);
+                    pendingMutations.onError(error);
                     subscriber.onError(error);
                 }
             );
         }));
     }
 
     /**
-     * Observe the {@link MutationOutbox}, for new {@link StorageItemChange}s.
-     * The Orchestrator may invoke this method to consume items out of the outbox. After
+     * Observe the {@link MutationOutbox}, for newly enqueued {@link PendingMutation}s.
+     * The {@link SyncProcessor} may invoke this method to consume items out of the outbox. After
      * processing an item on this observable, that item should be removed from the
-     * MutationOutbox.
+     * MutationOutbox via {@link #remove(PendingMutation)}.
      * @return An observable stream of items that have yet to be published via the network
      */
     @WorkerThread
     @NonNull
-    Observable<StorageItemChange<? extends Model>> observe() {
-        return pendingStorageItemChanges
+    Observable<PendingMutation<? extends Model>> observe() {
+        return pendingMutations
             .observeOn(Schedulers.io())
             .subscribeOn(Schedulers.io())
-            .startWith(previouslyUnprocessedChanges())
-            .filter(storageItemChange -> {
-                return !StorageItemChange.Initiator.SYNC_ENGINE.equals(storageItemChange.initiator());
-            });
```

Comment:
Just to make sure I'm understanding the removal here: we're saying that by virtue of SYNC_ENGINE-initiated records not being enqueued by the `StorageObserver`, this filter was essentially useless, right?

Reply:
Yes!

The `StorageObserver` also filters the `Initiator` right now. So by the time a model gets here, it had dang-sure already have been.

It should be possible for user models (`Blog`) to have the the API `Initiator` type. But these system models are only ever created internally, so they all have this internal `Initiator` type.